### PR TITLE
example/react-router-v4: Update push reference

### PR DESCRIPTION
**Pull request by Kikobeats**
**What project are you opening a pull request for?**
- react-instantsearch (use v2 base)

**Summary**

Related with https://github.com/algolia/instantsearch.js/pull/2023#issuecomment-284677368

### CODE REVIEW

**Comment by Kikobeats** [comment]
not sure if exists a way for check `history.push` is a `function`

File: packages/react-instantsearch/examples/react-router-v4/src/App.js [hunk 1]
```diff
@@ -91,7 +92,7 @@ class App extends Component {
 }
 
 App.propTypes = {
-  push: React.PropTypes.func.isRequired,
+  history: React.PropTypes.object.isRequired,
```
Comment (Haroenv) [inline packages/react-instantsearch/examples/react-router-v4/src/App.js]: http://stackoverflow.com/a/26924696/3185307

```js
{
  history: {
    push: React.PropTypes.func.isRequired,
  },
}
```

Comment (Kikobeats) [inline packages/react-instantsearch/examples/react-router-v4/src/App.js]: you're my hero

File: packages/react-instantsearch/examples/react-router-v4/src/App.js [hunk 2]
```diff
@@ -95,7 +95,7 @@ App.propTypes = {
   history: PropTypes.shape({
     push: PropTypes.func.isRequired,
   }),
-  location: PropTypes.object.isRequired
+  location: PropTypes.object.isRequired,
```
Comment (Kikobeats) [inline packages/react-instantsearch/examples/react-router-v4/src/App.js]: wut

Comment (Haroenv) [inline packages/react-instantsearch/examples/react-router-v4/src/App.js]: Sorry my fault

**Pull request merged**